### DISCUSSION TITLE
How do I run only specific misconfiguration scanners?

### DISCUSSION THREAD

**Discussion by alethenorio** [aquasecurity/trivy#4894] [Q&A] [answered]
### Question

I am trying to perform Terraform misconfiguration scanning in a repository which has a lot of other stuff than just HCL files.

What I am currently doing is `trivy config .` however trivy will also run kubernetes scanners when it finds yaml files.

I haven't been able to quite figure out from documentation how I can tell trivy to just run the Terraform scanners and nothing else. Is this possible?

### Target

None

### Scanner

Misconfiguration

### Output Format

Table

### Mode

Standalone

### Operating System

Linux

### Version

```bash
Version: 0.43.1
Vulnerability DB:
  Version: 2
  UpdatedAt: 2023-07-26 06:08:40.049602013 +0000 UTC
  NextUpdate: 2023-07-26 12:08:40.049601613 +0000 UTC
  DownloadedAt: 2023-07-26 09:42:21.671742071 +0000 UTC
Policy Bundle:
  Digest: sha256:acf1cb62a4ade75773ec3d265cc4d90160d285bbacaa7b6947a34830c1e45c5c
  DownloadedAt: 2023-07-31 10:04:41.624651257 +0000 UTC
```

**Comment by simar7** [comment]
You can limit by file extension and/or paths. https://aquasecurity.github.io/trivy/v0.43/docs/configuration/skipping/#advanced-globbing

**Reply by alethenorio** [reply]
Thank you. I'll double check this again but I think I tried this and as I understood from the docs (and my own experimentation) it will always look for every file type it knows.

So what I ended up with is having to supply a dummy value for every file type that is known to trivy that I don't want to match on.

For example something like `--file-patterns "dockerfile:.*.donotmatchonthis"` and hope no such file extension ever exists.

This feels more like a workaround than a proper solution.

Are you saying there is no other way of saying what misconfiguration module should or should not be run? Maybe an issue should be created then?

**Reply by simar7** [reply]
At the moment we don't have a way to disable certain misconfiguration analyzer selectively. We can open a feature request to do so.

**Reply by simar7** [reply, marked as answer]
Opened feature request here https://github.com/aquasecurity/trivy/issues/4901

**Reply by alethenorio** [reply]
Thank you
